### PR TITLE
Local backtrace generation using debuginfod

### PR DESCRIPTION
**Pull request by mgrabovsky**
- Instruct GDB to use Fedora's debuginfod server to pull necessary debuginfo files. The environment is [ready in Fedora 35+](https://fedoraproject.org/wiki/Changes/DebuginfodByDefault).
- Remove remote retracing capabilities completely. Drop `abrt-retrace-client` and `analyze_RetraceServer`.

### CODE REVIEW

**Comment by msrb** [comment]
One problem that I noticed while testing this PR is that there is a hardcoded 240 sec timeout limit in `abrt-action-generate-backtrace`. This was likely fine in the past as debuginfo packages were pre-downloaded on the laptop before we tried generating the backtrace. But now, we need to wait for gdb+debuginfod to fetch the missing data for us. All this needs to happen in under 240 seconds.

**Comment by msrb** [comment]
Q: do we want to keep the remote retracing as an option for now? Or rather cut it off completely?
I'd probably vote for the latter.

**Comment by mgrabovsky** [comment]
@msrb @michalfabik @mzidek-gh Calling for a last review. There's a [Copr build](https://copr.fedorainfracloud.org/coprs/g/abrt/devel/build/4838031/) if you're interested.

Note that this only works on Fedora 37+ due to the GLib dependency bump. I tested this on Fedora 37 and it seems to work fine in most cases – `sleep` and `evince` crashes give nice backtraces.

Though one problem I've noticed is that when retracing 1gnome-calendar1, for instance, GDB prints out a lot of warnings and doesn't seem to be able to load the relevant symbols for generating a useful backtrace:

    warning: Can't open file /usr/bin/gnome-calendar (deleted) during file-backed mapping note processing
    warning: Can't open file /memfd:mesa-shared (deleted) during file-backed mapping note processing
    [...]
    Thread 1 (LWP 2676):
    #0  0x00007fb3223d7d6f in ?? ()
    No symbol table info available.
    #1  0x000055c3e6143940 in ?? ()
    No symbol table info available.
    #2  0x000055c3e6143940 in ?? ()
    No symbol table info available.
    #3  0x0000000000000003 in ?? ()
    No symbol table info available.
    #4  0x000080e500000010 in ?? ()
    No symbol table info available.
    [...]

I have no idea if this is a configuration problem or a bug in GDB, but I recall we've encountered this in Retrace Server as well.

One last thing I just realised: This is most likely going to need an `Obsoletes: abrt-retrace-client <= x.y.z`, right?

**Comment by mgrabovsky** [comment]
The GitHub build is failing because the `fedora:latest` image is still 36, so we'll have to wait a month or so for that to go through.

**Comment by mzidek-gh** [comment]
"This is most likely going to need an Obsoletes: abrt-retrace-client <= x.y.z, right?"
Yes. But not just x.y.z , I think it should obsolete all versions of the package.

As for the missing debuginfo symbols. If you saw the same issue on retrace server I would say it is not related to this patch and you do not need to solve it in this PR.

**Comment by msrb** [comment]
@mgrabovsky progress indicator can be tricky. I think good old timeout should do the trick for now and we can figure out better way later :)

**Comment by msrb** [comment]
> "This is most likely going to need an Obsoletes: abrt-retrace-client <= x.y.z, right?"
> Yes. But not just x.y.z , I think it should obsolete all versions of the package.

I think Matej covers all previous versions with `<=` sign, or is there anything else that needs to be done?
https://docs.fedoraproject.org/en-US/packaging-guidelines/#renaming-or-replacing-existing-packages

**Comment by mzidek-gh** [comment]
> I think Matej covers all previous versions with `<=` sign, or is there anything else that needs to be done? https://docs.fedoraproject.org/en-US/packaging-guidelines/#renaming-or-replacing-existing-packages

Hmm... I though if you do not include the <= sign it will automatically mean all versions of that package. So I though it can be omitted. Maybe I was wrong. But of course setting to <= to include all version will also work, we can go with that as well.

File: src/plugins/analyze_LocalGDB.xml.in [hunk 1]
```diff
@@ -1,9 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <event>
     <_name>Local GNU Debugger</_name>
-    <_description>Download debuginfo packages and generate backtrace locally using GDB</_description>
-    <_long-description>Needs to downloads debuginfo packages, which might take significant time, and take up disk space.
-    However, unlike RetraceServer, doesn't send coredump to remote machines.
-    </_long-description>
+    <_description>Generate backtrace locally using GDB with debuginfod integration</_description>
+    <_long-description>Needs to downloads debuginfo files, which might take significant time, and take up disk space.</_long-description>
```
Comment (michalfabik) [inline src/plugins/analyze_LocalGDB.xml.in:5]: "Needs to download"

Comment (mgrabovsky) [reply src/plugins/analyze_LocalGDB.xml.in:5]: Good catch, thanks. Fixed now.

**Comment by mgrabovsky** [comment]
I've added the `Obsoletes` so this should be ready for merge if it's otherwise OK.

File: src/include/libabrt.h [hunk 2]
```diff
@@ -44,7 +44,7 @@ void abrt_ensure_writable_dir_uid_gid(const char *dir, mode_t mode, uid_t uid, g
 void abrt_ensure_writable_dir(const char *dir, mode_t mode, const char *user);
 void abrt_ensure_writable_dir_group(const char *dir, mode_t mode, const char *user, const char *group);
 char *abrt_run_unstrip_n(const char *dump_dir_name, unsigned timeout_sec);
-char *abrt_get_backtrace(struct dump_dir *dd, unsigned timeout_sec, const char *debuginfo_dirs);
+char *abrt_get_backtrace(struct dump_dir *dd, unsigned timeout_sec);
```
Comment (michalfabik) [inline src/include/libabrt.h:47]: One call to `abrt_get_backtrace` with the third arg still included is left in `abrt-action-generate-core-backtrace.c`.

Comment (mgrabovsky) [reply src/include/libabrt.h:47]: Good catch, fixed.

**Comment by michalfabik** [comment]
I think I have nothing to add, LGTM.

**Comment by mgrabovsky** [comment]
OK, thank you for your reviews. I'm going to merge this now. We can still debug any problems before a release.

**Pull request merged**